DEVELOPER_GUIDE.md

```diff
@@ -65,7 +65,7 @@ Download Java 11 from [here](https://adoptium.net/releases.html?variant=openjdk1
 
 #### JDK 8 and 17
 
-To run the full suite of tests, download and install [JDK 8](https://adoptium.net/releases.html?variant=openjdk8) and [17](https://adoptium.net/releases.html?variant=openjdk17) and set `JAVA8_HOME`, `JAVA11_HOME`, and `JAVA14_HOME`. They are required by the [backwards compatibility test](./TESTING.md#testing-backwards-compatibility).
+To run the full suite of tests, download and install [JDK 8](https://adoptium.net/releases.html?variant=openjdk8) and [17](https://adoptium.net/releases.html?variant=openjdk17) and set `JAVA8_HOME`, `JAVA11_HOME`, and `JAVA17_HOME`. They are required by the [backwards compatibility test](./TESTING.md#testing-backwards-compatibility).
```

Comment:
Changing what we have released feels wrong to me. Running bcw tests means using the JDK used for _that_ release, and that's unfortunately 14.

Reply:
I've updated the document to specify a [location to download JDK 14](https://jdk.java.net/archive/). Please let me know if there's a better place to link to. I've also removed reference to JDK17 and setting JAVA17_HOME because in my testing it is not required in order to get `./gradlew check` to pass (I may be missing something, but the only time I see JDK 17 being used is if I explicitly specify `-Druntime.java=17`)